### PR TITLE
rfct: group pages into new namespace

### PR DESCRIPTION
**Pull request by tangjeff0**
- Regroup page files into `athens.views.pages` namespace: see https://miro.com/app/board/o9J_kqT7am0=/?moveToWidget=3074457357149411251&cot=14

image

- Name the final export to be `page`
- Create an `athens.pages.page` that chooses between `block-page` and `node-page`
- Create an `athens.pages.core` that chooses page based on router
- Name the final export for some components to be the component's name, e.g. `app-toolbar/app-toolbar`

Old

```
src/cljs/athens/views
├── alerts.cljs
├── all_pages.cljs
├── app_toolbar.cljs
├── athena.cljs
├── block_page.cljs
├── blocks.cljs
├── breadcrumbs.cljs
├── buttons.cljs
├── daily_notes.cljs
├── data_browser.cljs
├── devtool.cljs
├── dropdown.cljs
├── filesystem.cljs
├── filters.cljs
├── graph_page.cljs
├── left_sidebar.cljs
├── modal.cljs
├── node_page.cljs
├── presence.cljs
├── right_sidebar.cljs
├── settings_page.cljs
├── spinner.cljs
├── textinput.cljs
└── toggle_switch.cljs
```

New

```
src/cljs/athens/views
├── pages
│   ├── all_pages.cljs
│   ├── block_page.cljs
│   ├── core.cljs
│   ├── daily_notes.cljs
│   ├── graph.cljs
│   ├── node_page.cljs
│   ├── page.cljs
│   └── settings.cljs
├── alerts.cljs
├── app_toolbar.cljs
├── athena.cljs
├── blocks.cljs
├── breadcrumbs.cljs
├── buttons.cljs
├── data_browser.cljs
├── devtool.cljs
├── dropdown.cljs
├── filesystem.cljs
├── filters.cljs
├── left_sidebar.cljs
├── modal.cljs
├── presence.cljs
├── right_sidebar.cljs
├── spinner.cljs
├── textinput.cljs
└── toggle_switch.cljs

```

### CODE REVIEW

**Comment by tangjeff0** [comment]
Curious if @shanberg and @neotyk think this hierarchy would make sense for the remaining components.

image

- `blocks` gets its own namespace (to decompose in another PR)
  - (can breakdown `node-page` in a separate PR as well)
- reused components are in a `components` namespace
- single-use components are at the top level of athens.views


```
src/cljs/athens/views
├── blocks
│   └── blocks.cljs
├── components
│   ├── alerts.cljs
│   ├── breadcrumbs.cljs
│   ├── buttons.cljs
│   ├── dropdown.cljs
│   ├── filesystem.cljs
│   ├── filters.cljs
│   ├── modal.cljs
│   ├── spinner.cljs
│   ├── textinput.cljs
│   └── toggle_switch.cljs
├── pages
│   ├── all_pages.cljs
│   ├── block_page.cljs
│   ├── core.cljs
│   ├── daily_notes.cljs
│   ├── graph.cljs
│   ├── node_page.cljs
│   ├── page.cljs
│   └── settings.cljs
├── app_toolbar.cljs
├── athena.cljs
├── data_browser.cljs
├── devtool.cljs
├── left_sidebar.cljs
├── presence.cljs
└── right_sidebar.cljs
```

**Comment by shanberg** [comment]
> Curious if @shanberg and @neotyk think this hierarchy would make sense for the remaining components.

Looks rational to me.

File: src/cljs/athens/core.cljs [hunk 1]
```diff
@@ -34,7 +34,7 @@
   []
   (rf/clear-subscription-cache!)
   (router/init-routes!)
-  (r-dom/render [views/main-panel]
+  (r-dom/render [views/view]
```
Comment (neotyk) [inline src/cljs/athens/core.cljs:37]: It feels like `main-panel` was a better name.

**Comment by neotyk** [comment]
> Curious if @shanberg and @neotyk think this hierarchy would make sense for the remaining components.

It might be easier to navigate, yes.

File: src/cljs/athens/core.cljs (hunk 1 above)
Comment (tangjeff0) [reply src/cljs/athens/core.cljs]: renamed to `main`. I just didn't understand what a `panel` as opposed to just a page.

**Comment by shanberg** [comment]
this seems like a good opportunity to rename "left sidebar" to "navigation" and "right sidebar" to "references". If we want to treat them as very abstract containers I think "left/right sidebar" is fine, but at the moment the sidebars are specific to those purposes.

**Comment by tangjeff0** [comment]
@shanberg right sidebar also can have graph view. Haven't mentioned this yet, but do you think left sidebar could have other actions beyond just shortcuts? Some of the stuff in our toolbar could be put in the left sidebar, might reduce toolbar busy-ness.

image

image

**Comment by shanberg** [comment]
> Haven't mentioned this yet, but do you think left sidebar could have other actions beyond just shortcuts? Some of the stuff in our toolbar could be put in the left sidebar, might reduce toolbar busy-ness.

Sounds like a good topic for discussion

**Comment by shanberg** [comment]
> Name the final export for some components to be the component's name, e.g. app-toolbar/app-toolbar

Is there no index.js equivalent in cljs?

**Comment by tangjeff0** [comment]
Not sure, thoughts @neotyk?

**Pull request merged**